Adding content for the new flexible consent option (#2167)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/api/oidc/index.md

A consent dialog appears depending on the values of three elements:

> * `prompt` - a query parameter used in requests to [`/authorize`](/docs/reference/api/oidc/#authorize)

Reviewer: ```suggestion
* `prompt` - a query parameter that is used in requests to [`/authorize`](/docs/reference/api/oidc/#authorize)
```

> * `consent_method` - an [application](/docs/reference/api/apps/#settings-7) property. Allows you to determine whether a client is fully trusted (for example: a first-party application) or requires consent (for example: a third-party application).

Reviewer: ```suggestion
* `consent_method` - an [application](/docs/reference/api/apps/#settings-7) property. Allows you to determine whether a client is fully trusted (for example, a first-party application) or requires consent (for example, a third-party application).
```

* `consent` - a property on Scopes as listed in the table above. Allows you to individually enable and disable user consent for a scope.

| `prompt` Value   | `consent_method`        | `consent`                            | Result       |
| :--------------- | :---------------------- | :----------------------------------- | :----------- |
| `CONSENT`        | `TRUSTED` or `REQUIRED` | `REQUIRED`                           | Prompted     |
| `CONSENT`        | `TRUSTED` or `REQUIRED` | `FLEXIBLE`                           | Prompted     |
| `CONSENT`        | `TRUSTED`               | `IMPLICIT`                           | Not prompted |
| `NONE`           | `TRUSTED`               | `FLEXIBLE`, `IMPLICIT`, or `REQUIRED`| Not prompted |
| `NONE`           | `REQUIRED`              | `FLEXIBLE` or `REQUIRED`             | Prompted     |
| `NONE`           | `REQUIRED`              | `IMPLICIT`                           | Not prompted |

> > **Note:** If `CONSENT` is set to `FLEXIBLE`, and the scope is requested when using the Client Credentials grant flow, the scope is granted in the access token with no consent prompt. This is because there is no user involved in a 2-legged OAuth [Client Credentials](/docs/guides/implement-client-creds/overview/) grant flow.

Reviewer: ```suggestion
> **Note:** If `CONSENT` is set to `FLEXIBLE` and the scope is requested when using the Client Credentials grant flow, the scope is granted in the access token with no consent prompt. This occcurs because there is no user involved in a two-legged OAuth [Client Credentials](/docs/guides/implement-client-creds/overview/) grant flow.
```
